test/integration/targets/git/tasks/separate-git-dir.yml

```diff
@@ -12,12 +12,21 @@
   register: tempdir
 
 - name: SEPARATE-GIT-DIR | clone with a separate git dir
-  command: git clone {{ repo_format1 }} {{ checkout_dir }} --separate-git-dir={{ tempdir.stdout }}
+  git:
+    repo: '{{ repo_format1 }}'
+    dest: '{{ checkout_dir }}'
+    separate_git_dir: '{{ tempdir.stdout }}'
 
 - name: SEPARATE-GIT-DIR | update repo the usual way
   git:
     repo: "{{ repo_format1 }}"
     dest: "{{ checkout_dir }}"
+  register: result
```

Comment:
And what if you change it to some other separate directory? Test and fix that.

Reply:
And what should happen when you completely remove this option?

Reply:
What if it was cloned without this option and then it appeared?

Reply:
Please test/design all these cases.

Reply:
For now, if we run git the first time, git module will do a clone. Once a repo is cloned, the git module will just do an update (pull) but not clone when you call it. So it doesn't matter whether I remove the option or add the option, it will not call clone again. Is this ok? Or I add the code that checks the position of separate git dir, and if the position changes I relocate the separate git dir?

Reply:
It is inconsistent: if the user changes this option, they probably expect some change. I think we should discuss this question in slack. We should think-through all possible transitions of repo state.

Also, let's ping @cmrd-senya @giacmir @robinro @iprok @drybjed  to hear their opinions as they primarily raised the question regarding support of this mode.